lib/digest.rb

```diff
@@ -8,7 +12,7 @@ module Digest
   def self.const_missing(name) # :nodoc:
     case name
     when :SHA256, :SHA384, :SHA512
-      lib = 'digest/sha2.so'
+      lib = RUBY_ENGINE == 'jruby' ? 'digest/sha2' : 'digest/sha2.so'
```

Comment:
Would simply `lib = 'digest/sha2'` work?

Reply:
(That was my "question to myself", but I didn't dare ask it.)

Reply:
yes, it's simpler, thanks for the suggestion.